pkg/webhook/utils/utils.go

```diff
@@ -201,5 +180,69 @@ func ParseNetworkConfigOfPodByPriority(ctx context.Context, c client.Reader, pod
 		}
 	}
 
+	networkType = ipamtypes.ParseNetworkTypeFromString(networkTypeStr)
+	if len(ipFamily) == 0 {
+		ipFamily = ipamtypes.ParseIPFamilyFromString(ipFamilyStr)
+	}
+
+	if len(networkName) > 0 {
+		network := &networkingv1.Network{}
+		if err = c.Get(ctx, types.NamespacedName{Name: networkName}, network); err != nil {
+			err = fmt.Errorf("failed to get network %v: %v", networkName, err)
+			return
+		}
+
+		// specified network takes higher priority than network type defaulting, if no network type specified
+		// from pod, then network type should inherit from network type of specified network from pod
+		if len(networkTypeStr) == 0 {
+			networkType = ipamtypes.ParseNetworkTypeFromString(string(networkingv1.GetNetworkType(network)))
+		}
```

Comment:
maybe it's need to check the conflict between network-name and network-type

Reply:
done